src/fu-engine.c

```diff
@@ -724,6 +725,136 @@ fu_engine_modify_remote(FuEngine *self,
 	return fu_remote_list_set_key_value(self->remote_list, remote_id, key, value, error);
 }
 
+static gboolean
+fu_engine_validate_bios_attr_input(FwupdBiosAttr *self, const gchar *value, GError **error)
+{
+	guint64 tmp;
+
+	switch (fwupd_bios_attr_get_kind(self)) {
+	case FWUPD_BIOS_ATTR_KIND_READ_ONLY:
```

Comment:
i'm not sure the switch is providing very much here; maybe just `if()` -- and then you can use the correct `gsize tmp` for the `strlen`.